setuptools/dist.py

```diff
@@ -636,6 +637,17 @@ def dash_to_underscore_warning(self, opt, section):
                 % (opt, underscore_opt))
         return underscore_opt
 
+    def uppercase_warning(self, opt, section):
+        if section in ('metadata',) and (any(c.isupper() for c in opt)):
+            lowercase_opt = opt.lower()
+            warnings.warn(
```

Comment:
This should probably have a `DeprecationWarning` argument too.

Reply:
In my experience, `DeprecationWarnings` are suppressed by default so aren't useful for user-facing warnings. Perhaps a `SetuptoolsDeprecationWarning` would be appropriate, but if I recall correctly, the other warning issued is the default (`UserWarning`). If a different warning class is preferred, let's do that in a separate change.

Reply:
@jaraco pyca/cryptography does this and it causes problems for people attempting to granularly ignore the thing when the place where it's emitted is in the import chain of the custom exception.

See: https://github.com/pytest-dev/pytest/issues/7524 / https://github.com/pyca/cryptography/issues/5335 / https://github.com/pyca/cryptography/pull/5346 / https://cryptography.io/en/3.4.6/faq.html#i-cannot-suppress-the-deprecation-warning-that-cryptography-emits-on-import